dipy/reconst/mcsd.py

```diff
@@ -0,0 +1,347 @@
+        iso: int (optional)
+            Number of tissue compartments for running the MSMT-CSD. Minimum
+            number of compartments required is 2.
+            Default: 2
+
+        References
+        ----------
+        .. [1] Jeurissen, B., et al. NeuroImage 2014. Multi-tissue constrained
+               spherical deconvolution for improved analysisof multi-shell
+               diffusion MRI data
+        .. [2] Tournier, J.D., et al. NeuroImage 2007. Robust determination of
+               the fibre orientation distribution in diffusion MRI:
+               Non-negativity constrained super-resolved spherical
+               deconvolution
+        .. [3] Tournier, J.D, et al. Imaging Systems and Technology
+               2012. MRtrix: Diffusion Tractography in Crossing Fiber Regions
+        """
+
+        sh_order = response.sh_order
+        super(MultiShellDeconvModel, self).__init__(gtab)
+        B, m, n = multi_tissue_basis(gtab, sh_order, iso)
+
+        delta_f = delta_functions[delta_form]
+        delta = delta_f(response.iso, response.m, response.n, 0., 0.)
+        self.delta = delta
+        multiplier_matrix = _inflate_response(response, gtab, n, delta)
+
+        r, theta, phi = geo.cart2sphere(*reg_sphere.vertices.T)
+        odf_reg, _, _ = shm.real_sym_sh_basis(sh_order, theta, phi)
+        reg = np.zeros([i + iso for i in odf_reg.shape])
+        reg[:iso, :iso] = np.eye(iso)
+        reg[iso:, iso:] = odf_reg
+
+        X = B * multiplier_matrix
+
+        self.fitter = QpFitter(X, reg)
+        self.sh_order = sh_order
+        self._X = X
+        self.sphere = reg_sphere
+        self.response = response
+        self.B_dwi = B
+        self.m = m
+        self.n = n
+
+    def predict(self, params, gtab=None, S0=None):
+        """Compute a signal prediction given spherical harmonic coefficients
+        for the provided GradientTable class instance.
+
+        Parameters
+        ----------
+        params : ndarray
+            The spherical harmonic representation of the FOD from which to make
+            the signal prediction.
+        gtab : GradientTable
+            The gradients for which the signal will be predicted. Use the
+            model's gradient table by default.
+        S0 : ndarray or float
+            The non diffusion-weighted signal value.
+            Default : None
+        """
+        if gtab is None:
+            X = self._X
+        else:
+            iso = self.response.iso
+            B, m, n = multi_tissue_basis(gtab, self.sh_order, iso)
+            multiplier_matrix = _inflate_response(self.response, gtab, n,
+                                                  self.delta)
+            X = B * multiplier_matrix
+        return np.dot(params, X.T)
+
+    @multi_voxel_fit
+    def fit(self, data):
+        coeff = self.fitter(data)
+        return MSDeconvFit(self, coeff, None)
+
+
+class MSDeconvFit(shm.SphHarmFit):
```

Comment:
This one needs some documentation

Reply:
Also, is the ODF inherited directly from `SphHarmFit` correct?

Reply:
> Also, is the ODF inherited directly from `SphHarmFit` correct?

Yes :+1:

Reply:
Great. It still needs some documentation.